pegasus/helpers/form_helpers.rb

```diff
@@ -129,17 +131,19 @@ def insert_form(kind, data, options={})
 
   timestamp = DateTime.now
 
+  normalized_email = data[:email_s].to_s.strip.downcase
   row = {
     secret: SecureRandom.hex,
     parent_id: options[:parent_id],
-    email: data[:email_s].to_s.strip.downcase,
+    email: normalized_email,
     name: data[:name_s].to_s.strip,
     kind: kind,
     data: data.to_json,
     created_at: timestamp,
     created_ip: request.ip,
     updated_at: timestamp,
     updated_ip: request.ip,
+    hashed_email: Digest::MD5.hexdigest(normalized_email),
```

Comment:
Actually, I'm flipping on this - looking at all the places I'd have to touch to make this work, it's a decent number. I think it's safest for the moment to leave the field as possibly null (that's the current state in the DB), and handle the process of enforcing not-null later. To do so now risks causing actual errors, whereas the only fallout from having some null hashed_emails that could be set is that this feature will miss a "hit", which is less problematic. This is currently the only feature that uses this field at all, so making sure we tolerate null on the read-side is easy.

Once we're populating the hashed_email in production, we'll be able to see if there are still records showing up without them, and I'll be a more confident about flipping the switch and migrating the DB table to enforce it.